dockerd-patches/snappy-buildkit-git-environ.patch

```diff
@@ -1,22 +1,14 @@
-https://github.com/docker-snap/docker-snap/pull/25#issuecomment-827849995
-https://github.com/moby/buildkit/pull/1533/files#diff-9a3248027714dcbcf2647fa27b75ec4dd3d9639ae80c40bfd6a68abd7805bd4eR494-R498
-
 diff --git a/vendor/github.com/moby/buildkit/source/git/gitsource.go b/vendor/github.com/moby/buildkit/source/git/gitsource.go
-index 3d1bfe21f4..197c339c78 100644
+index 15e0d56a21..d94075a3ca 100644
 --- a/vendor/github.com/moby/buildkit/source/git/gitsource.go
 +++ b/vendor/github.com/moby/buildkit/source/git/gitsource.go
-@@ -598,12 +598,11 @@ func git(ctx context.Context, dir, sshAuthSock, knownHosts string, args ...strin
- 		cmd.Stdin = nil
- 		cmd.Stdout = io.MultiWriter(stdout, buf)
+@@ -613,6 +613,9 @@ func git(ctx context.Context, dir, sshAuthSock, knownHosts string, args ...strin
  		cmd.Stderr = io.MultiWriter(stderr, errbuf)
--		cmd.Env = []string{
--			"PATH=" + os.Getenv("PATH"),
-+		cmd.Env = append(append([]string{}, os.Environ()...), []string{
```

Comment:
Did you leave all of `os.Environ()` in here on purpose?

(there's probably no need to do the later `os.Getenv` calls if so, but I think this list was intentionally limited to mitigate different environment variables that `git` might pull in that would affect the way buildkit invokes it :grimacing:)

Reply:
@tianon it's a diff of a diff
```console
- +		cmd.Env = append(append([]string{}, os.Environ()...), []string{
```
meaning the line of append all of os.Environ, is removed from this patch.

see https://github.com/xnox/docker-snap/blob/7a5e055ffdf5c48efc5ea3a1884c79e98ceaf17b/dockerd-patches/snappy-buildkit-git-environ.patch how the patch actually looks like now.

Reply:
Doh, yes, sorry; it has been a long week. :facepalm:

Thank you for explaining :bow: